core/blockchain.go

```diff
@@ -3158,6 +3219,147 @@ func (bc *BlockChain) IsEnablePruneBeaconChainFeature() bool {
 	return bc.pruneBeaconChainEnable
 }
 
+// SyncFromTiKVWriter used for tikv mode, all reader or follower writer used to sync block from master writer
+func (bc *BlockChain) SyncFromTiKVWriter(newBlkNum uint64, logs []*types.Log) error {
+	head := rawdb.ReadHeadBlockHash(bc.db)
+	dbBlock := bc.GetBlockByHash(head)
+	currentBlock := bc.CurrentBlock()
+
+	if dbBlock == nil || currentBlock == nil {
+		return nil
+	}
+
+	currentBlockNum := currentBlock.NumberU64()
+	if currentBlockNum < newBlkNum {
+		start := time.Now()
+		for i := currentBlockNum; i <= newBlkNum; i++ {
+			blk := bc.GetBlockByNumber(i)
+			if blk == nil {
+				// todo check later
+				return nil
+			}
+
+			err := bc.tikvFastForward(blk, logs)
+			if err != nil {
+				return err
+			}
+		}
+		log.Printf("sync to %d use %v", currentBlockNum, time.Now().Sub(start))
+	}
+
+	return nil
+}
+
+// tikvCleanCache used for tikv mode, clean block tire data from redis
+func (bc *BlockChain) tikvCleanCache() {
+	var count int
+	for to := range bc.cleanCacheChan {
+		for i := bc.latestCleanCacheNum + 1; i <= to; i++ {
+			// build previous block statedb
+			fromBlock := bc.GetBlockByNumber(i)
+			fromTrie, err := state.New(fromBlock.Root(), bc.stateCache)
+			if err != nil {
+				continue
+			}
+
+			// build current block statedb
+			toBlock := bc.GetBlockByNumber(i + 1)
+			toTrie, err := state.New(toBlock.Root(), bc.stateCache)
+			if err != nil {
+				continue
+			}
+
+			// diff two statedb and delete redis cache
+			start := time.Now()
+			count, err = fromTrie.DiffAndCleanCache(bc.ShardID(), toTrie)
+			if err != nil {
```

Comment:
won't you return the error here? Just a log message is enough?

Reply:
Failing to clean the cache doesn't affect the process in any way